### ISSUE TITLE
Issue with fullRandom?

### ISSUE THREAD

**Issue by tpronk** [psychopy/psychojs#134]
This time I know how fullRandom should work :), but I'm worried that it might not work as expected in PsychoJS. In the experiment below, part "random_funky", I randomise a sequence of 1s and 2s. Given a long enough sequence, I would expect to see, at some point, three 1s or three 2s in a row. With the given random seed I see this behavior in PsychoPy, but not in PsychoJS. I replicated this with values of nReps > 3 and a different seed.
- https://gitlab.pavlovia.org/tpronk/e2e_conditions
- https://run.pavlovia.org/tpronk/e2e_conditions/html/

I'd like to use this issue for writing a little unit test, fix the bug (if any), and make a pull-request. For bonus points, see if we can align the random number generators of PsychoPy and PsychoJS.

**Comment by tpronk** [comment]
I made a (very ugly) unit test. See the README in [this repo](https://github.com/tpronk/test_random) for an explanation.

TLDR: all tests were passed, except one that compared the output of a FULL_RANDOM randomisation via the PsychoJS library with e2e_conditions: random_funky. My guess is that something goes wrong on another level?

**Comment by tpronk** [comment]
If found the problem. 

1. When generating code from the builder using a loop with fullRandom, PsychoPy generates... `method: TrialHandler.Method.FULLRANDOM`
2. There is no symbol for this constant, so its value defaults to `undefined`
3. As a consequence, in the TrialHandler constructor, the default value for the method argument is used, which is `TrialHandler.Method.RANDOM`, see [this line](https://github.com/psychopy/psychojs/blob/da2d9c5ad720beb2de9ae6a93d9c4a5dfbc07ebb/js/data/TrialHandler.js#L69)
4. Hence, it assumes random and is not caught by the else-clause on [this line](https://github.com/psychopy/psychojs/blob/da2d9c5ad720beb2de9ae6a93d9c4a5dfbc07ebb/js/data/TrialHandler.js#L638)

Solutions:
1. Rename `TrialHandler.Method.FULL_RANDOM` to `TrialHandler.Method.FULLRANDOM`
2. No default value for method argument

**Comment by peircej** [comment]
Good hunting @tpronk 
That was a chewy one! ;-)

**Comment by tpronk** [comment]
Yeah, this one was fun :)
